recipes/uriparser/all/conanfile.py

```diff
@@ -62,6 +71,7 @@ def _patch_sources(self):
                                   "")
 
     def build(self):
+        self._patch_sources()
```

Comment:
I don't see the point of this patch, without it, URI_PUBLIC is defined as __declspec(dllimport) when consuming uriparser as a shared lib on msvc, which is fine, or I've missed something.

Reply:
It will only remove `__declspec(dllimport)` when building a static uriparser library.

This is done to avoid this warning:
https://docs.microsoft.com/en-us/cpp/error-messages/tool-errors/linker-tools-warning-lnk4217?view=vs-2019

Reply:
If static, URI_STATIC_BUILD is defined (when building, but also when consuming through cpp_info), therefore URI_PUBLIC can't be __declspec(dllimport)

In CMakeLists.txt:
```cmake
if (NOT BUILD_SHARED_LIBS)
    target_compile_definitions(uriparser PUBLIC URI_STATIC_BUILD)
endif()
```

In UriBase.h:
```c++
#if defined(URI_STATIC_BUILD)
# define URI_PUBLIC
#else
# if defined(URI_LIBRARY_BUILD)
#  if defined(_MSC_VER)
#   define URI_PUBLIC __declspec(dllexport)
#  elif defined(URI_VISIBILITY)
#   define URI_PUBLIC __attribute__ ((visibility("default")))
#  else
#   define URI_PUBLIC
#  endif
# else
#  if defined(_MSC_VER)
#   define URI_PUBLIC __declspec(dllimport)
#  else
#   define URI_PUBLIC
#  endif
# endif
#endif
```

Reply:
You're right. I'll remove the patch.